### PR TITLE
Branch update user guide

### LINKED ISSUE #132: Update User Guide to v1.4

### PR DESCRIPTION
**Pull request by Green-Tea-123**
closes #132 
This updates the user guide to cover all the features we have implemented till v1.4's release.
It does not have any features implemented since then and also does not have any future features (supposed to labelled coming soon).

### CODE REVIEW

**Review by wassilililily — APPROVED** [review]
LGTM

File: docs/UserGuide.md [hunk 1]
```diff
@@ -65,81 +69,223 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
 
 ### Viewing help : `help`
 
-Shows a message explaning how to access the help page.
+Shows a message explaining how to access the help page.
```
Comment (Quasant) [inline docs/UserGuide.md:72]: nice catch

File: docs/UserGuide.md [hunk 2]
```diff
@@ -65,81 +69,223 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
...
+Undoes a matching between a contact and job.
+
+Format: `unmatch CONTACT_INDEX JOB_INDEX`
+
+* Undoes the matching of a contact at the specified `CONTACT_INDEX` and a job
+at the specified `JOB_INDEX`.
+* The contact index refers to the index number shown in the displayed contact list.
+* The job index refers to the index number shown in the displayed job list.
+* Both indices **must be positive integers** 1, 2, 3, …​
+* The contact and job **must already be matched together**.
```
Comment (Quasant) [inline docs/UserGuide.md:264]: This is fine, but as mentioned in the tele chat I will be making it such that job does not keep a reference anymore, so the last line in the match and unmatch might need to be amended to something like `Contact must not already by matched`.  

Earliest I will be able to do the change is on Monday, so apologies for that.

**Review by Quasant — APPROVED** [review]
Overall, things look good.

File: docs/UserGuide.md (hunk 2 above)
Comment (Green-Tea-123) [reply docs/UserGuide.md:264]: Actually, maybe you could edit the UG after you change that to register that you edited the UG for this milestone.

**Pull request merged**